### DISCUSSION TITLE
Example of cascading combobox

### DISCUSSION THREAD

**Discussion by satyajit-behera** [microsoft/fluentui-blazor#3038] [Q&A] [answered]
Hi, is there any example of cascading combobox.
Ex - If I select country, based on that, states should be filled.
Thanks in advance.

**Comment by vnbaaij** [comment]
See https://github.com/microsoft/fluentui-blazor/issues/1298 for example code with Select. Same idea

**Reply by satyajit-behera** [reply]
Thanks. Is there a way to make it work for ComboBox:

 ```
<FluentCombobox Autocomplete="ComboboxAutocomplete.Inline" Height="200px" Items=@CountryOptions
                     OptionText="@(i => i.Text)"
                     OptionValue="@(i => i.Value)"
                     OptionSelected="@(i => i.Selected)"
                     ValueChanged="HandleCountryChanged" />
                     
  private async Task HandleCountryChanged(string? value)
  {
      if (value == null)
      {
          return;
      }
      SelectedCountry = value;
      await FillStates(value);
  
    }
```

 This does not work.

**Reply by satyajit-behera** [reply]
One key challenge with ValueChanged="HandleCountryChanged" is that it takes the binding text as the value. Difficult to get the binding value which is usually the Id and need to be passed for next call.

**Reply by vnbaaij** [reply]
You cannot rely on the value to be there in case of a Combobox. The user is free to input anything into the text part and then **there is no** corresponding value

**Reply by satyajit-behera** [reply]
Oh ok

**Reply by satyajit-behera** [reply]
I tried it with some setter and events but that does not seem to be predictable. Is there a way we can implement search for FluentSelect also. Thanks.

**Reply by vnbaaij** [reply]
Nope. For search you'd need to use the autocomplete

**Reply by satyajit-behera** [reply]
Thanks

**Reply by satyajit-behera** [reply]
while using autocomplete, is it possible to get the underlying Pk value of the selected items. ValueChanged="HandleCountryChanged" gives only the typed value in corresponding method param.

**Reply by vnbaaij** [reply, marked as answer]
I would advise to use `@bind-SelectedItems` and then add a `@bind-SelectedItems:after` method to work with the variable you used to store the selected items in. You can get to all the item data that way. You shouldn't have to tie in to the binding process like you are trying to do.

See https://www.fluentui-blazor.net/Autocomplete#default for an example. We just get the country name and display it, but you could get all `Country` properties like that

**Reply by satyajit-behera** [reply]
Thanks a lot. Its awesome. When an item is **removed** from the list, it does not fire the method for `@bind-SelectedItems:after` . can you pls guide on that.

**Reply by vnbaaij** [reply]
If you use the cross on the items' presentation in the autocomplete, it will be removed from the selection. You can see in the example I linked to that the display of selected countries gets updated. I don't think there is a way to trigger the `:after` callback like that

**Reply by satyajit-behera** [reply]
Its working great now. Thanks a lot. Kudos to the entire team for coming up with these great ui components.
